infra/templates/dns-records/template.yaml

```diff
@@ -0,0 +1,94 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: >-
+  Creates the necessary components to manage DNS in the Async ID Check journey and the logging for the Route 53 service.
+
+Mappings:
+  PlatformConfiguration:
+    dev:
+      DNSSUFFIX: review-b-async.dev.account.gov.uk
+    build:
+      DNSSUFFIX: review-b-async.build.account.gov.uk
+    staging:
+      DNSSUFFIX: review-b-async.staging.account.gov.uk
+    integration:
+      DNSSUFFIX: review-b-async.integration.account.gov.uk
+    production:
+      DNSSUFFIX: review-b-async.account.gov.uk
+
+Parameters:
+  Environment:
+    Description: The name of the environment to deploy to
+    Type: String
+    AllowedValues:
+      - dev
+      - build
+      - staging
+      - integration
+      - production
+
+Conditions:
+  IsDevEnvironment: !Equals
+    - !Ref Environment
+    - dev
+
+Resources:
+  ExternalCertificate:
+    Type: AWS::CertificateManager::Certificate
+    Properties:
+      DomainName: !Sub
+      - "${DNSSUFFIX}"
+      - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+
+      SubjectAlternativeNames:
+      - !Sub
+        - "api.${DNSSUFFIX}"
+        - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+      - !If
+        - IsDevEnvironment  
+        - !Sub
+          - "*.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+        - !Ref AWS::NoValue
+
+      DomainValidationOptions:
+        - DomainName: !Sub
+          - "${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+          HostedZoneId: !ImportValue PublicHostedZoneId
+        - DomainName: !Sub
+          - "api.${DNSSUFFIX}"
+          - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+          HostedZoneId: !ImportValue PublicHostedZoneId        
+      ValidationMethod: DNS
+      Tags:
+        - Key: Name
+          Value: !Sub
+            - "${DNSSUFFIX}"
+            - DNSSUFFIX: !FindInMap [ PlatformConfiguration, !Ref Environment, DNSSUFFIX ]
+        - Key: Product
+          Value: "ID Check Async"
+        - Key: System
+          Value: "ID Check Async"
+        - Key: Environment
+          Value: !Sub "${Environment}"
+
+  CertificateARNSSM:
```

Comment:
I'm tempted to suggest we remove this parameter. 

I've had a brief discussion with @sandymay-dd about how and when to use parameters verses exports verses hardcoding values. We never settled on any guidelines so I welcome your input on what you'd like to do in this case.

Here I'm considering how we might want to access the certificate arn. 

When using it in another cloudformation stack then using the export prevents us from deleting the cloudformation stack until all usages of it is gone. I don't think happens with a parameter. 

On the other hand a parameter is valuable when retrieving the value outside of cloudformation, so potentially used when configuring a containers environment variables for example, or accessing the value as part of a lambda run.

Reply:
Actually, ignore all of the above. Looking through the existing code base the certificate is only accessed through the parameter. I trust they gave this decision more thought than me so I think we should stick to that approach.